In [1]:
import mlflow
import numpy as np

# A/B test sample data
group_a = np.array([1, 0, 1, 1, 0, 1, 0, 1, 1, 0])
group_b = np.array([1, 1, 1, 1, 0, 1, 1, 0, 1, 1])

# Calculate conversion rates
conversion_a = group_a.mean()
conversion_b = group_b.mean()

# Calculate difference
difference = conversion_b - conversion_a

# Start MLflow experiment
mlflow.set_experiment("W9D3_AB_Test")

with mlflow.start_run() as run:

    mlflow.log_param("test_type", "A/B Test")
    mlflow.log_param("sample_size_A", len(group_a))
    mlflow.log_param("sample_size_B", len(group_b))

    mlflow.log_metric("conversion_rate_A", conversion_a)
    mlflow.log_metric("conversion_rate_B", conversion_b)
    mlflow.log_metric("conversion_difference", difference)

    better_group = "B" if conversion_b > conversion_a else "A"
    mlflow.log_param("better_group", better_group)

    print("A/B Test completed successfully")
    print("Group A Conversion Rate:", conversion_a)
    print("Group B Conversion Rate:", conversion_b)
    print("Difference:", difference)
    print("Better Group:", better_group)
    print("Run ID:", run.info.run_id)

2026/10/01 13:24:04 INFO mlflow.tracking.fluent: Experiment with name 'W9D3_AB_Test' does not exist. Creating a new experiment.


A/B Test completed successfully
Group A Conversion Rate: 0.6
Group B Conversion Rate: 0.8
Difference: 0.20000000000000007
Better Group: B
Run ID: 6df4a59348ab4112a6ff5208d95fdda2


# W9D3 - A/B Test Tracking with MLflow

## Objective
The objective was to perform a simple A/B test and track the results using MLflow.

## Process
- Created sample data for Group A and Group B.
- Calculated the conversion rate for both groups.
- Calculated the difference between the conversion rates.
- Logged the A/B test parameters and metrics in MLflow.
- Compared both groups to identify the group with the higher conversion rate.

## Results
- Group A Conversion Rate: 0.60
- Group B Conversion Rate: 0.80
- Conversion Difference: 0.20
- Better Group based on conversion rate: B

## MLflow Tracking
The experiment was created as `W9D3_AB_Test`.
The MLflow Run ID was:
`6df4a59348ab4112a6ff5208d95fdda2`

## Conclusion
Group B had a higher conversion rate than Group A in this sample A/B test. The experiment results were successfully tracked using MLflow.